### DISCUSSION TITLE
Network check

### DISCUSSION THREAD

**Discussion by yelloguy** [tteck/Proxmox#1337] [General]
Instead of doing 10 network checks every 3 seconds, can we do 3 every 180 seconds? Or make it configurable in advanced? All your scripts are failing for me on network check. However, when I change network checks to 3 every 180 seconds, it fails on the first one and succeeds on the second one.
There are others reporting "no network" errors too.

**Comment by tteck** [comment]
Why is your router so slow at assigning an IP?

**Reply by tteck** [reply]
A 3 minute pause in between checks is unacceptable for 99.99% of users.

**Reply by yelloguy** [reply]
If I test the network from the console of the CT I just spun up right after the network check fails, I can ping Google, and Github. But the script fails.

It could be taking 3x10 = 30 seconds. But I don't mind a 180 second delay since the scripts take long anyway. And that is why I proposed configuration prompt in advanced.

**Reply by yelloguy** [reply]
Also the delay is after first fail. If someone has the first check succeed, they get 0 delay

**Reply by tteck** [reply]
My recommendation would be to resolve the problem with your network rather than modifying the script to accommodate the issue.

**Reply by tteck** [reply]
Are you using IPv6?

**Reply by yelloguy** [reply]
Thanks! I would if I knew what the problem is and how to resolve it.

For now, I am running your docker script with changed delay (180x3) and this is what I got. After the (3x10) script failed 

```
+ echo -en '\033[01;31m✗\033[m\033[01;31m No Network! '
✗ No Network! + sleep 180
+ (( i-- ))
+ (( i > 0 ))
++ hostname -I
+ '[' '192.168.0.190 ' '!=' '' ']'
+ break
+ msg_ok 'Set up Container OS'
+ local 'msg=Set up Container OS'
+ echo -e '\r\033[K \033[1;92m✓\033[m \033[1;92mSet up Container OS\033[m'
 ✓ Set up Container OS
++ hostname -I
+ msg_ok 'Network Connected: \033[36m192.168.0.190 '
+ local 'msg=Network Connected: \033[36m192.168.0.190 '
+ echo -e '\r\033[K \033[1;92m✓\033[m \033[1;92mNetwork Connected: \033[36m192.168.0.190 \033[m'
 ✓ Network Connected: 192.168.0.190 
+ network_check
+ set +e
+ trap - ERR
+ ping -c 1 -W 1 1.1.1.1
+ msg_ok 'Internet Connected'
+ local 'msg=Internet Connected'
+ echo -e '\r\033[K \033[1;92m✓\033[m \033[1;92mInternet Connected\033[m'
 ✓ Internet Connected
++ awk '{ print $1 }'
++ getent hosts github.com
+ RESOLVEDIP=140.82.113.4
+ [[ -z 140.82.113.4 ]]
+ msg_ok 'DNS Resolved github.com to \033[36m140.82.113.4\033[m'
+ local 'msg=DNS Resolved github.com to \033[36m140.82.113.4\033[m'
+ echo -e '\r\033[K \033[1;92m✓\033[m \033[1;92mDNS Resolved github.com to \033[36m140.82.113.4\033[m\033[m'
 ✓ DNS Resolved github.com to 140.82.113.4
```

**Reply by yelloguy** [reply]
No to IPv6. It is disabled on my router and I say "no" to the script/advanced

**Reply by yelloguy** [reply]
to be honest I am fine with this. I am just suggesting the change for others who might have the same problem. If you don't like 3x180, you can up that interval from 10x3 to something like 5x10 or even 5x20, or something. Seems like a simple fix.
